packages/webview_flutter/lib/webview_flutter.dart

```diff
@@ -385,6 +395,7 @@ class _WebSettings {
 class WebViewController {
```

Comment:
is the long term plan to get rid of this class in favor of WebViewPlatform?

Reply:
This is the public API we expose to our users(vs the public `WebViewPlatform` which is the API we expose to platform implementors), while the public API for these 2 currently matches I'm not sure we should require it to match.

Also note that renaming WebViewController right now will be a breaking change, I guess we could eventually make WebViewController extend WebViewPlatform if we want to save one indirection layer, and we extract all the utility code(like the logic in `_updateJavaScriptChannels`) out of it.

Reply:
I agree that this should not be part of this PR, but in general I think this controller and WebViewPlatform going to look fairly similar so it would be nice if we can unify them

Reply:
Let's re-discuss it after we move all the methodchannel stuff out of WebViewController?

I'm actually not sure which option I prefer, I can imagine having a value in separating the interface that users are using to control the webview from the interface that platforms need to provide for the plugin to control the webview (e.g I can see us adding some logic in WebViewController that don't belong in the platform implementation).
But I do see your point about the code duplication.

We should decide at the end of this PR series before publishing to avoid a breaking change.